### PR TITLE
feat: add contributing section

### PR DESCRIPTION
**Pull request by mfrancisc**
This PR moves the content ( with some slight modifications ) from https://github.com/codeready-toolchain/toolchain-e2e/blob/master/dev_install.adoc

### CODE REVIEW

File: docs/quay-repo-config.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    - `https://quay.io/repository/<quay-username>/host-operator-index`
+    - `https://quay.io/repository/<quay-username>/member-operator`
+    - `https://quay.io/repository/<quay-username>/member-operator-webhook`
+    - `https://quay.io/repository/<quay-username>/member-operator-bundle`
+    - `https://quay.io/repository/<quay-username>/member-operator-index`
+    - `https://quay.io/repository/<quay-username>/registration-service`
+
+### Public visibility
+
+All aforementioned repositories has to be public, so make sure that the visibility is set to `public` for all of them:
```
Comment (rsoaresd) [inline docs/quay-repo-config.md]:
```suggestion
All aforementioned repositories have to be public, so make sure that the visibility is set to `public` for all of them:
```

**Review by rsoaresd — COMMENTED** [review]
Looks good!! Just minor suggestion:

File: docs/quay-repo-config.md (hunk 1 above)
Comment (mfrancisc) [reply docs/quay-repo-config.md]: TY!

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+### OpenShift Cluster
+Ensure you have access to an OpenShift 4.6+ cluster with cluster admin privileges and log in using `oc login`
+
+### Required Tools
+Install the [required tools](required-tools.md).
+
+### Authentication
+Configure authentication for the cluster using one of the following options:
+
+1. Contact a member of the Dev Sandbox Team for instructions on how to configure the cluster to use our internal Dev SSO.
```
Comment (alexeykazakov) [inline docs/contributing.md]: Maybe
```suggestion
1. Contact a member of the KubeSaw Team for instructions on how to configure the cluster to use our internal Dev SSO.
```
???

**Review by alexeykazakov — APPROVED** [review]
Nice!

File: docs/contributing.md (hunk 2 above)
Comment (mfrancisc) [reply docs/contributing.md]: yeah I was in doubt here, but makes sense, just for consistency. thanks!

File: docs/contributing.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+### OpenShift Cluster
+Ensure you have access to an OpenShift 4.6+ cluster with cluster admin privileges and log in using `oc login`
+
+### Required Tools
+Install the [required tools](required-tools.md).
+
+### Authentication
+Configure authentication for the cluster using one of the following options:
+
+1. Contact a member of the KubeSaw Team for instructions on how to configure the cluster to use our internal Dev SSO.
```
Comment (metlos) [inline docs/contributing.md]: Given that we're aiming for Kubesaw to be installable by the 3rd parties, I think this should not be mentioned as the 1st option. We could leave it in as the last option and prepend it with something like "If you're a Redhatter, you can also ...".

File: docs/contributing.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+oc patch clusterrolebinding.rbac self-provisioners -p '{"subjects": null, "metadata": {"annotations":{"rbac.authorization.kubernetes.io/autoupdate": "false"}}}'
+oc adm policy remove-cluster-role-from-group self-provisioner system:authenticated:oauth
+```
+
+### Install KubeSaw
+Clone this repository `git clone git@github.com:codeready-toolchain/toolchain-e2e.git`
+
+This repository provides you multiple Makefile targets that you can use - it depends on which version of KubeSaw operators you want to install.
+
+**NOTE**: If the cluster is an OSD cluster, then set the variable `IS_OSD=true` when running any of the Makefile targets (for example: `make appstudio-dev-deploy-latest IS_OSD=true`).
```
Comment (metlos) [inline docs/contributing.md]: I think `make dv-deploy-latest IS_OSD=true` would be better here as AppStudio is most probably something the general public is going to be interested in using.

File: docs/contributing.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+### Latest greatest KubeSaw for AppStudio
+Run the following to install the latest greatest KubeSaw operators in dev mode for AppStudio environment:
+```
+make appstudio-dev-deploy-latest
+```
+
+### Local version
+If you want to install a local version of a specific KubeSaw operator in dev mode then:
```
Comment (metlos) [inline docs/contributing.md:62]: This assumes certain directory layout or additional config, I think. Shouldn't we mention it here, too?

File: docs/contributing.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+1. _kube:admin_
+2. The authentication method configured in the [Authentication step](#authentication)
+![Log in](https://user-images.githubusercontent.com/20015929/114628295-a141eb00-9c84-11eb-8be3-45f013e19378.png)
+Select option 2 and log in using the same account used from the [Register a KubeSaw User section](#register-a-new-kubesaw-user) step.
+
+After logging in, the user will have access to only the namespaces created for them.
+
+### Cleanup
+#### Remove Only Users and Their Namespaces
```
Comment (metlos) [inline docs/contributing.md:142]: Maybe mention here that the make targets are available in the e2e repo?

File: docs/contributing.md (hunk 3 above)
Comment (mfrancisc) [reply docs/contributing.md]: Good point! I'll do that. Even if TBH this is for now just an INTERNAL Contribution guide. I would expect that we have a complete new/different one once we open for external contributions.

File: docs/contributing.md (hunk 4 above)
Comment (mfrancisc) [reply docs/contributing.md]: makes sense

File: docs/contributing.md (hunk 3 above)
Comment (mfrancisc) [reply docs/contributing.md]: done in: https://github.com/kubesaw/kubesaw.github.io/pull/10/commits/f7fc24c415cfc0f65cf0a928707b8fc569578319

File: docs/contributing.md (hunk 5 above)
Comment (mfrancisc) [reply docs/contributing.md:62]: good point, added more details in https://github.com/kubesaw/kubesaw.github.io/pull/10/commits/f7fc24c415cfc0f65cf0a928707b8fc569578319

File: docs/contributing.md (hunk 4 above)
Comment (mfrancisc) [reply docs/contributing.md]: fixed in https://github.com/kubesaw/kubesaw.github.io/pull/10/commits/f7fc24c415cfc0f65cf0a928707b8fc569578319

File: docs/contributing.md (hunk 6 above)
Comment (mfrancisc) [reply docs/contributing.md:142]: good point, done in https://github.com/kubesaw/kubesaw.github.io/pull/10/commits/f7fc24c415cfc0f65cf0a928707b8fc569578319

**Review by metlos — APPROVED** [review]

**Pull request merged**